app/src/main/java/com/pajato/android/gamechat/fragment/TTTFragment.java

```diff
@@ -3,47 +3,84 @@
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.Button;
 import android.widget.TextView;
 
+import com.firebase.client.DataSnapshot;
+import com.firebase.client.Firebase;
+import com.firebase.client.FirebaseError;
+import com.firebase.client.GenericTypeIndicator;
+import com.firebase.client.ValueEventListener;
 import com.pajato.android.gamechat.R;
 import com.pajato.android.gamechat.game.GameManager;
 
+import java.util.HashMap;
 import java.util.Scanner;
 
 public class TTTFragment extends Fragment {
 
     private static final String TAG = TTTFragment.class.getSimpleName();
+    private static final String FIREBASE_URL = "https://gamechat-1271.firebaseio.com/boards/ticTacToe";
+    private static final String TURN_INDICATOR = "turnIndicator";
+
     /* Keeps track of the Turn user. True = Player 1, False = Player 2. */
     public boolean mTurn;
 
-    // Keeps track of the turn number
-    private int mTurnCount;
-
     // Player Piece Strings
     private String mXValue;
     private String mOValue;
+    private String mSpace;
 
-    // The mBoard view
+    // Board management objects
     private View mBoard;
+    private HashMap<String, Integer> mBoardMap;
+    private Firebase mRef;
 
     public TTTFragment() {
-
+        // Required empty constructor.
     }
 
-    @Override public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                                       Bundle savedInstanceState) {
+    @Override public View onCreateView(final LayoutInflater inflater, final ViewGroup container,
+                                       final Bundle savedInstanceState) {
+        // Initialize Member Variables
+        mBoard = inflater.inflate(R.layout.fragment_ttt, container, false);
         mXValue = getString(R.string.xValue);
         mOValue = getString(R.string.oValue);
+        mSpace = getString(R.string.spaceValue);
         mTurn = true;
 
-        // Initialize the turn counter, and mBoard variables.
-        mTurnCount = 9;
-        mBoard = inflater.inflate(R.layout.fragment_ttt, container, false);
+        // Setup our Firebase database reference and a listener to keep track of the board.
+        Firebase.setAndroidContext(getContext());
+        mRef = new Firebase(FIREBASE_URL);
+        mRef.addValueEventListener(new ValueEventListener() {
+            @Override
+            // If the data changes on Firebase, we want to capture the new board and replicate it locally.
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                GenericTypeIndicator<HashMap<String, Integer>> t =
+                        new GenericTypeIndicator<HashMap<String, Integer>>() {};
+                mBoardMap = dataSnapshot.getValue(t);
+                // If there is a board to be had, fill our board out with it and adjust the turn.
+                if(mBoardMap != null) {
+                    recreateExistingBoard();
+                    checkNotFinished();
```

Comment:
This reminds me that a game/activity has state, such as "active" or "completed".